homework_basic_components/src/task8/Task8.java

```diff
@@ -0,0 +1,166 @@
+package task8;
+
+import java.awt.BorderLayout;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JPasswordField;
+import javax.swing.JTextField;
+
+/**
+ * Represents small GUI application for registration.
+ * Registration will be successful if username and password 
+ * are longer than 6 characters, and if password contains at least
+ * one special character and one number.
+ * @author ajla
+ *
+ */
+public class Task8 extends JFrame {
+
+	private static final long serialVersionUID = 1788825113584408633L;
+
+	// Declaring variables
+	private JPanel pnlMain = new JPanel();
+	private JPanel pnlTextFields = new JPanel();
+	private JPanel pnlButtons = new JPanel();
+
+	private JButton btnClear = new JButton("Clear");
+	private JButton btnSubmit = new JButton("Submit");
+
+	private JTextField tfUsername = new JTextField();
+	private JPasswordField pfPassword = new JPasswordField();
+	private JPasswordField pfReenterPassword = new JPasswordField();
+
+	public Task8() {
+
+		// Main panel layout
+		pnlMain.setLayout(new BorderLayout());
+		pnlMain.add(pnlTextFields, BorderLayout.CENTER);
+
+		// TextFields panel layout
+		pnlTextFields.setLayout(new GridLayout(3, 2));
+
+		// Adding text fields to main panel
+		pnlTextFields.add(new JLabel("Username: "));
+		pnlTextFields.add(tfUsername);
+		pnlTextFields.add(new JLabel("Password: "));
+		pnlTextFields.add(pfPassword);
+		pnlTextFields.add(new JLabel("Reenter password: "));
+		pnlTextFields.add(pfReenterPassword);
+
+		// Buttons panel position
+		pnlMain.add(pnlButtons, BorderLayout.SOUTH);
+
+		// Adding button
+		pnlButtons.add(btnClear);
+		pnlButtons.add(btnSubmit);
+
+		// Adding main panel to window
+		add(pnlMain);
+
+		// Window settings
+		setSize(300, 180);
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setLocationRelativeTo(null);
+		setTitle("Task 8");
+		setVisible(true);
+
+		// Adding ActionListener to Submit button
+		btnSubmit.addActionListener(new Submit());
+
+		// Adding ActionListener to Clear button
+		btnClear.addActionListener(new ActionListener() {
+
+			public void actionPerformed(ActionEvent e) {
+				if (e.getSource() == btnClear) {
+					tfUsername.setText("");
+					pfPassword.setText("");
+					pfReenterPassword.setText("");
+				}
+			}
+		}); 
+	}
+
+	public static void main(String[] args) {
+		new Task8();
+	}
+
+	private class Submit implements ActionListener {
+
+		public void actionPerformed(ActionEvent e) {
+			if (e.getSource() == btnSubmit) {
+
```

Comment:
Consider creating separate methods for validating user/pwd. It will make it much more easy to maintain and action listener itself would be much more readable.